### DISCUSSION TITLE
Building Prior Parameter Covariance Matrix with >35k parameters?

### DISCUSSION THREAD

**Discussion by kes79** [gmdsi/GMDSI_notebooks#109] [Q&A]
In the PstFrom notebook, it is possible to build the prior parameter covariance matrix and save as a binary file, and this becomes the basis for building the prior ensemble.  However, I have too many parameters to do this? What is the trickery mentioned below that I need to use in this case please?

"For cases with less than about 30,000 parameters, we can actually generate and visualize the prior parameter covariance matrix. If you have more parameters, this matrix may not fit in memory. But, not to worry, PstFrom has some trickery to help generate the geostatistical prior ensemble."

**Comment by jtwhite79** [comment]
This is a common question and the answer to it is mentioned just below the prior covariance matrix block in the notebook:

<img width="860" alt="image" src="https://github.com/gmdsi/GMDSI_notebooks/assets/1770912/24541d1f-9488-4c8a-9042-2285d7681117">

The `PstFrom.draw()` method internally forms and processes the prior covariance matrix block-by-block (that is parameter group by parameter group) to generate the prior parameter ensemble.  

I'm thinking we might want to add more flashing red lights about this issue and the prior covariance matrix bit in the notebook - we only included the prior covariance matrix for demonstration...
